pkg/operator/controllers/pullsecret/pullsecret_controller_test.go

```diff
@@ -761,13 +780,20 @@ func TestEnsureGlobalPullSecret(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			ctx := context.Background()
 
+			var kubernetesFake *fake.Clientset
 			clientBuilder := ctrlfake.NewClientBuilder()
 			if tt.initialSecret != nil {
 				clientBuilder = clientBuilder.WithObjects(tt.initialSecret)
+				kubernetesFake = serversideapply.CliWithApply([]string{"secrets"}, tt.initialSecret)
+			} else {
+				kubernetesFake = serversideapply.CliWithApply([]string{"secrets"})
```

Comment:
Can this be simplified? I'm wondering if an empty `tt.initialSecret` is treated the same as `tt.initialSecret` not being supplied to `CliWithApply` that way we don't need the else branch